portal/components/table/_components/virtualRows.tsx

```diff
@@ -0,0 +1,58 @@
+import { flexRender, Row } from '@tanstack/react-table'
+
+import { useTableVirtualizer } from '../_hooks/useTableVirtualizer'
+
+import { Column } from './column'
+
+type Props<TData> = {
+  getRowKey?: (row: TData, index: number) => string
+  loading: boolean
+  rows: Row<TData>[]
+  virtualItems: ReturnType<
+    ReturnType<typeof useTableVirtualizer<TData>>['getVirtualItems']
+  >
```

Comment:
Similar to https://github.com/hemilabs/ui-monorepo/pull/1520/files#r2359712561 see if there's a type provided by `@tanstack/react-virtual` that works for us